packages/router/router.ts

```diff
@@ -242,6 +244,16 @@ export interface Router {
    */
   deleteBlocker(key: string): void;
 
+  /**
+   * @internal
+   * PRIVATE DO NOT USE
+   *
+   * Patch additional children routes into an existing parent route
+   * @param routeId The parent route id
+   * @param children The additional children routes
+   */
+  patchRoutes(routeId: string | null, children: AgnosticRouteObject[]): void;
```

Comment:
Used by Remix to eagerly/imperatively patch routes in _prior_ to navigations

Reply:
Why is this considered private? Wouldn't this be an important part of the public API for a feature like this?

Reply:
Only for advanced pre-fetching use cases - not the common use case where you'll use `patch` from `patchRoutesOnMiss`.  This is more so just to stay in line with the current public API of `router` which is that it's all marked private and we prefer you use the React Router APIs in the vast majority of cases.  In v7 we won't have a standalone `@remix-run/router` package but we will want to start documenting the true "public" APIs of the `router`